proptest-derive/src/derive.rs

```diff
@@ -540,11 +540,7 @@ fn variant_no_explicit_strategy<C>(
             ctor
         } else {
             let pref = acc.add_param(params_ty);
-            if pref + 1 == count {
-                ctor
-            } else {
-                extract_all(ctor, count, FromReg::Num(pref))
-            }
+            extract_all(ctor, count, FromReg::Num(pref))
```

Comment:
I don't exactly recall why this `if` expression was there (Note to self: leave comments damn it!), but it seems to work since you have tests. =P

Reply:
@Centril, indeed, I was trying to grasp it myself. When I added introspection into the first branch of the `if`, it was being hit only in the tests that I've just added. I suppose it was originally meant to be an optimization of some sort but that's all I could figure. :)